tools/example_api_lab.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python3
+"""
+KernelCI API example
+3rd party lab submission, when lab is not integrated with KernelCI pipeline
+
+"""
+
+import requests
+import json
+import sys
+
+# KernelCI API URL
+api_url = "https://staging.kernelci.org:9000/latest"
+
+
+def add_authorization_header(headers):
+    # read api token from .api_token file
+    try:
+        with open(".api_token", "r") as f:
+            api_token = f.read().strip()
+    except FileNotFoundError:
+        print("Error: API token file not found")
+        sys.exit(1)
+    # Add the authorization header to the request
+    headers["Authorization"] = f"Bearer {api_token}"
+    return headers
+
+
+def retrieve_node(node_id):
+    url = api_url + "/node/" + node_id
+    headers = {"Content-Type": "application/json"}
+    response = requests.get(url, headers=headers)
+    if response.status_code != 200:
+        print("Failed to retrieve node:", response.text)
+        return None
+    return response.json()
+
+
+def create_node(node):
+    # Create a new node in the database
+    url = api_url + "/node"
+    headers = {"Content-Type": "application/json"}
+    headers = add_authorization_header(headers)
+    response = requests.post(url, headers=headers, data=json.dumps(node))
+    if response.status_code != 200:
+        print("Failed to create node:", response.text)
+        return None
+    try:
+        return response.json()
+    except json.JSONDecodeError:
+        print("Failed to create node:", response.text)
+        return None
+
+
+def update_node_tree(node_id, treedata):
+    # Update the node with a new tree
+    url = api_url + "/nodes/" + node_id
+    headers = {"Content-Type": "application/json"}
+    headers = add_authorization_header(headers)
+    print(treedata)
+    response = requests.put(url, headers=headers, data=json.dumps(treedata))
+    if response.status_code != 200:
+        print("Failed to update node tree:", response.text)
+        return None
+    return response
+
+
+def create_job(jobname, parent_node):
+    job_node = {
+        "kind": "job",
+        "name": jobname,
+        "path": ["checkout", jobname],
+        "group": jobname,
+        "parent": parent_node["id"],
+        "state": "running",
+        "result": None,
+        "artifacts": {},
+        "data": {},
+        "submitter": parent_node["submitter"],
+        "treeid": parent_node["treeid"],
+    }
+    job_node["data"]["kernel_revision"] = parent_node["data"]["kernel_revision"]
+    return create_node(job_node)
+
+
+def job_results(job_node_id, tests, artifacts):
```

Comment:
maybe call this `submit_job_results()`. to make it clear?

Reply:
done